### PR TITLE
#887  Exclude not active crumbs siblings

### LINKED ISSUE #887: Exclude non active crums siblings
It's screen from seo guys
image

### PR DESCRIPTION
**Pull request by duker33**
Closes #887

### CODE REVIEW

File: shopelectro/tests/tests_views.py [hunk 1]
```diff
@@ -527,20 +531,33 @@ def test_tags_pagination_has_canonical_links(self):
             )
         )
 
-    def test_category_matrix_page(self):
```
Comment (artemiy312) [inline shopelectro/tests/tests_views.py]: It looks like an accidentally removed test

File: shopelectro/tests/tests_views.py [hunk 2]
```diff
@@ -527,20 +531,33 @@ def test_tags_pagination_has_canonical_links(self):
...
-
-        categories_db = models.Category.objects.active()
-        categories_app = soup.find_all(tag='a', class_='catalog-list-item')
-        for from_db, from_app in zip(categories_db, categories_app):
-            self.assertEqual(from_db.name, from_app.a.text)
+    # @todo #887:30m  Fix not active crumbs siblings bug.
+    #  The test below proves the bug. Now fix it.
+    @unittest.expectedFailure
+    def test_crumb_siblings_are_active(self):
+        parent = models.Category.objects.raw(
```
Comment (artemiy312) [inline shopelectro/tests/tests_views.py]: It seems it is the same, but simpler 
```suggestion
        parent = models.Category.objects.annotate(c=Count('children')).filter(c__gt=1).first()
```

File: shopelectro/tests/tests_views.py [hunk 3]
```diff
@@ -527,20 +531,33 @@ def test_tags_pagination_has_canonical_links(self):
...
+        )[0]
+        (
+            pages_models.Page.objects
+            .filter(id=parent.children.first().page.id)
+            .update(is_active=False)
+        )
+        category = parent.children.active().first()
+        soup = self.get_category_soup(category.children.active().first())
+        siblings = soup.select('.breadcrumbs-siblings-links a')
+        self.assertTrue(all([
```
Comment (artemiy312) [inline shopelectro/tests/tests_views.py]: Up to you
```suggestion
        self.assertFalse(models.Category.objects.filter(
            name__in=[s.text.strip() for s in siblings],
            page__is_active=False,
        ).exists())
```

File: shopelectro/tests/tests_views.py [hunk 4]
```diff
@@ -20,7 +21,7 @@
 from django.utils.translation import ugettext as _
 
 from catalog.helpers import reverse_catalog_url
-from pages.models import CustomPage
+from pages import models as pages_models
```
[the remark this answers is not in the document]
Comment (duker33) [reply shopelectro/tests/tests_views.py]: @artemiy312 , some django models preprocessing black magic does not allow us to use just `pages.models` appropriately. I'll create subtask about it for research and possible renaming all over the system

File: shopelectro/tests/tests_views.py (hunk 1 above)
Comment (duker33) [reply shopelectro/tests/tests_views.py]: @artemiy312 , it's just moved to `shopelectro.tests.tests_views.CategoriesMatrix#test_roots_sorting` with another task. I just removed code old doubling

File: shopelectro/tests/tests_views.py (hunk 2 above)
Comment (duker33) [reply shopelectro/tests/tests_views.py]: thanks! I'll try it

**Pull request merged**